### PR TITLE
x64: Port fdemote and fvdemote to ISLE

### PR DESCRIPTION
**Pull request by elliottt**
For the x86_64 backend, port the `fdemote` and `fvdemote` instructions. Additionally, add tests to exercise those instructions as well as `fpromote` and `fvpromote_low`.

### CODE REVIEW

File: cranelift/codegen/src/isa/x64/lower.rs [hunk 1]
```diff
@@ -1033,21 +1033,11 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
...
 
         Opcode::Fvdemote => {
-            let src = RegMem::reg(put_input_in_reg(ctx, inputs[0]));
-            let dst = get_output_reg(ctx, outputs[0]).only_reg().unwrap();
-            ctx.emit(Inst::xmm_unary_rm_r(
-                SseOpcode::Cvtpd2ps,
-                RegMem::from(src),
-                dst,
-            ));
+            implemented_in_isle(ctx);
```
Comment (fitzgen) [inline cranelift/codegen/src/isa/x64/lower.rs:1040]: Ideally we would just move these opcodes up to the top of the match where the rest of the implemented-in-ISLE opcodes are all matched together.

**Review by fitzgen — APPROVED** [review]
Looks like we are missing run tests for `fdemote` and completely missing *any* file tests whatsoever for `fvdemote`. Care to add some of those along with this PR?

File: cranelift/codegen/src/isa/x64/lower.rs (hunk 1 above)
Comment (elliottt) [reply cranelift/codegen/src/isa/x64/lower.rs:1040]: I'll move them up to the top -- I was matching the surrounding style, figuring that we'd do a pass over this later.

**Comment by elliottt** [comment]
> Looks like we are missing run tests for `fdemote` and completely missing _any_ file tests whatsoever for `fvdemote`. Care to add some of those along with this PR?

Yep, happy to add tests for these. I was working on getting good test cases for fdemote/fpromote before marking this ready, as I noticed they were a little under tested.

**Pull request merged**